File: locks/typescript/test/catalog.test.mjs
import assert from "node:assert/strict";
import { test } from "node:test";
import { ORG, catalog, entryKey, entryPlan, key } from "../dist/index.js";

test("keys carry the org prefix", () => {
  assert.equal(key("jobs", "x"), "fanwaave/jobs/x");
  assert.equal(ORG, "fanwaave");
});

test("placeholders are filled in order and every entry plans", () => {
  assert.equal(entryKey({ domain: "jobs", name: "{a}/x/{b}", layers: { fiducia: true, pgAdvisory: true }, pgScope: "transaction", wait: true }, "1", "2"), "fanwaave/jobs/1/x/2");
  for (const entry of Object.values(catalog)) assert.ok(entryPlan(entry).steps.includes("work"));
});

File: locks/dart/test/catalog_test.dart
import 'package:fanwaave_locks/fanwaave_locks.dart';
import 'package:test/test.dart';

void main() {
  test('keys carry the org prefix', () {
    expect(key(Domain.jobs, 'x').value, 'fanwaave/jobs/x');
  });

  test('placeholders are filled in order', () {
    const entry = Entry(
      domain: Domain.jobs,
      name: '{a}/x/{b}',
      layers: LockLayers.both,
      pgScope: PgScope.transaction,
      wait: true,
    );
    expect(entry.key(['1', '2']).value, 'fanwaave/jobs/1/x/2');
    expect(entry.plan.steps.first, LockStep.fiduciaAcquire);
  });
}
